lesson35/link-on-playgtound.md

```diff
@@ -0,0 +1,5 @@
+1 lesson task 
+https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgMIHsCu4rAgZwHkAjAKwgTGQG8AoZB5BLHATwC5l8xcQBzADT1GcYsSgQAbsDhhg6EJ269BwhgmBgOXHqFWMmmKBJAJWAfRBwAthCW7+QgwAd0zzABtZ8xchCZrYmhaAF9aWmYQbiYWXQJODGw4ojIKMABtAF1kAF5kdLUaQoNmJKhtACIAVRBNCAATZABBKFFkAFFrYFbIfAqnAxExCWlvBU4Kpvb+4sYNLQmm4kxkABEAC1FgGcG5oxMzSxs7ZEnW4g6unoI17s3rHd3kV3cvOXHkAE4ANgBmAAZPgBWT6FEIDBh0J4xMqVAAK6C8IHqj12onEUhk718FThABlUYN5pUAOpwKD4ABecAA7nBCSV9hBTBYrLYJgiPMB8OtkAAtDzoLQMxgvTxjXy-ADsnylABYpUD-mCIUVoaU2BMAEqYfD4GQgZAAMQa0AlIoY6JGWJ82qqFqYmkqAFl0PhmDSHQgmSyjuzTjq9QbkDriB4IA6xW9bcgAIxyhNSgAc-yVYNomVoAHos7HkLQYNhKD4YWw8dywAAKclQTgtVqsAA81FLPG0yj0yBCAD4AJRqxgSMBGQ01gB01jgzkrEHD1ly3eQs4g1jHGrbvbHpHQoErAHIBMg972ANyhcI5gBMBaL2OQYCFcA8cIgbnD1eMdeMcCbLajEs4fxAmgLs+wHBghxHZBx0nacAA8F2QOCx3-bFNwkepMCQStqwQBBD29Yx+xyRdEAQZAAGpDGMQ9-lPc9syzX4b1MO98HQKAwESHA8HwD9a2ab9f1bcp7BUUD+yhQcIGHKBR2MCcp0rBCSKQtdYnKTd2M4yt6LCRi5RY4sFGokx8FYesf34r8G2bUzmUONkTg7fgJPA5BILk6CFNg5TEOQwiDlZY49IvLMgSMu9iQAZQ4sAmndazBNsltiTEzse0kwpPPkqBFPg-y1ydLS4t0s99Jzb580LViSzgSQzT4CAEVeCUkssptuOSEhyEoMCpIgmSoJgpSVMXZDUJ8dCGiwiAcLIgj9mI0i8Mo0zaP7LNvLy8N+DAdZyvCRiQCFGJjDSLhEUwbFaFASBYEQFAuvKDr3PXUSdBUBjImid6LM-NANP+hsslyfJCgGokgY5RE4GRCoVQh2Z1GhgNdX1OHjVNHofARgwwkyM9IrqyQ+Ba8VsQAIVYZ7WH42MEg03ierSLJDxrS9GdhDqsiykoFGiGtCBgZ1ZAQdYCGe3iwZrWMxxgYAPHu+mAElIHnEjke2y8x3Y2xKy1gwObVldENlk3Vz+3IcjyY31fU2Etd7Qpnf5qIqBrPFmT4faZeMYXRbAcXJaZggx12n31myoavKFkWxYl-ApbD3yxrUyaFGmzDsNw-DTOW6DVqowKNuQLbPe9-bzyAA
```

Comment:
```typescript
//5 
function carsList(arr: Array<IUsers>) {
    return arr.filter(user => user.cars).map(x => x.cars).join();
}
```
calling `join` on nested arrays is not obvious so better to use `flat` at the end or `flatMap`